docs/docs/terraform/main.tf

```diff
@@ -0,0 +1,466 @@
+}
+
+data "aws_vpc" "neptune_contained_herein" {
+  id = local.vpc_id
+}
+
+data "aws_subnet_ids" "neptune_contained_herein" {
+  vpc_id = data.aws_vpc.neptune_contained_herein.id
+}
+
+resource "aws_security_group" "allow_gremlin" {
+  count       = 1
+  name        = "allow_gremlin"
+  description = "allow inbound to gremlin"
+  vpc_id      = data.aws_vpc.neptune_contained_herein.id
+}
+
+resource "aws_security_group_rule" "allow_gremlin_ipv4" {
+  count           = length(aws_security_group.allow_gremlin)
+  type            = "ingress"
+  from_port       = 8182
+  to_port         = 8182
+  protocol        = "tcp"
+  prefix_list_ids = []
+  # we assume the VPC is isolated
+  cidr_blocks = [
+    "0.0.0.0/0"
+  ]
+  ipv6_cidr_blocks = []
+  security_group_id = aws_security_group.allow_gremlin[count.index].id
+}
+
+resource "aws_security_group_rule" "allow_gremlin_ipv6" {
+  count           = length(aws_security_group.allow_gremlin)
+  type            = "ingress"
+  from_port       = 8182
+  to_port         = 8182
+  protocol        = "tcp"
+  prefix_list_ids = []
+  # we assume the VPC is isolated
+  ipv6_cidr_blocks = [
+    "::/0"
+  ]
+  cidr_blocks = []
+  security_group_id = aws_security_group.allow_gremlin[count.index].id
+}
+
+resource "aws_security_group" "allow_outbound" {
+  count       = 1
+  name        = "allow_outbound"
+  description = "allow all outbound"
+  vpc_id      = data.aws_vpc.neptune_contained_herein.id
+}
+
+resource "aws_security_group_rule" "allow_outbound_ipv4" {
+  count           = length(aws_security_group.allow_outbound)
+  type            = "egress"
+  protocol        = "all"
+  from_port       = 0
+  to_port         = 0
+  prefix_list_ids = []
+  cidr_blocks = [
+    "0.0.0.0/0"
+  ]
+  ipv6_cidr_blocks = []
+  security_group_id = aws_security_group.allow_outbound[count.index].id
+}
+
+resource "aws_security_group_rule" "allow_outbound_ipv6" {
+  count           = length(aws_security_group.allow_outbound)
+  type            = "egress"
+  protocol        = "all"
+  from_port       = 0
+  to_port         = 0
+  prefix_list_ids = []
+  ipv6_cidr_blocks = [
+    "::/0"
+  ]
+  cidr_blocks = []
+  security_group_id = aws_security_group.allow_outbound[count.index].id
+}
+
+data "aws_caller_identity" "current" {}
+
+# neptune things
+
+resource "aws_neptune_cluster_parameter_group" "service" {
+  family = "neptune1" # magic, but currently the only family
+  description = var.service
+  name = var.service
+
+  parameter {
+    name = "neptune_enable_audit_log"
+    value = 1
+  }
+
+  parameter {
+    name = "neptune_enforce_ssl"
+    value = 1
+  }
+}
+
+resource "aws_neptune_subnet_group" "service" {
+  name       = "service"
+  subnet_ids = data.aws_subnet_ids.neptune_contained_herein.ids
+}
+
+resource "aws_neptune_cluster" "service" {
+  count = 1
+  engine = "neptune"
+  cluster_identifier = var.service
+  iam_database_authentication_enabled = true
+  # engine_version default
```

Comment:
Is this configuration syntax that's been commented out, or a comment explaining something? If the former maybe delete it outright. If the later, needs more context.

Reply:
Looks like [syntax](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/neptune_cluster#engine_version) commented out, IIUC you can pick which release of neptune you want. It's not super clear to me why you'd want a specific engine version, so I'll probably remove.